### ISSUE TITLE
verilator does not detect if block name is used twice

### ISSUE THREAD

**Issue by veripoolbot** [verilator/verilator#485]
---

Author Name: **Alex Solomatnikov**
Original Redmine Issue: 485 from https://www.veripool.org
Original Date: 2012-04-19

---

Code:

```
    always_comb begin
       integer q_ind;

       for( q_ind=0; q_ind<NUM_QUEUES-1; q_ind++ ) begin: COMB
...
       for( q_ind=0; q_ind<NUM_QUEUES; q_ind++ ) begin: COMB

```

verilator compiles without complaint.

VCS errors out:

```
Error-[IPD] Identifier previously declared
  Identifier 'COMB' previously declared as NamedBlock.
  Source info:       for( q_ind=0; q_ind<NUM_QUEUES; q_ind++ ) begin: COMB

```

**Comment by ldoolitt** [comment]
I turned the snippet above into a stupid little test case, attached.
[issue_485.zip](https://github.com/verilator/verilator/files/8189957/issue_485.zip)
The issue is indeed present with v4.010 as shipped in Debian Buster.
But the error is properly detected with v4.038 as shipped in Debian Bullseye (source date: Jul 11 2020).
Also good on several more recent versions I checked: v4.102, v4.110, v4.202, v.214, v4.216-120-g956f64c6b.
Can this be closed?  Or does someone need to run the git bisect, to find the commit that fixed it?
